routes/user.signup.js

```diff
@@ -79,6 +79,9 @@ router.post('/', function(req, res) {
             password: password
         });
 
+        //this temp user will be used to check whether the username is already exist or not
+        //this has to be done because nev.createTempUser from email-verification api just handle the account
```

Comment:
So this is how we know if a user exist and will error if the username already exists? I guess it's not clear to me why e-mail-verification is related to this.